File: rumahsehat-mobile/lib/models/user_model.dart
class User {
  String jwtToken;
  String id;
  String nama;
  String username;
  String role;
  String email;

  User({
    required this.jwtToken,
    this.id = "",
    this.nama = "",
    this.username = "",
    this.role = "",
    this.email = "",
  });

  User.fromJson(Map<String, dynamic> json)
      : jwtToken = json['jwtToken'],
        id = json['id'],
        nama = json['nama'],
        username = json['username'],
        role = json['role'],
        email = json['email']
        ;

  Map<String, dynamic> toJson() => {
        'jwtToken': jwtToken,
        'id': id,
        'nama': nama,
        'username': username,
        'role': role,
        'email': email,
      };
}

File: rumahsehat-website/src/main/java/apap/tugaskelompok/rumahsehat/model/UserModel.java
package apap.tugaskelompok.rumahsehat.model;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import javax.persistence.*;
import javax.persistence.Entity;
import javax.validation.constraints.*;

import org.hibernate.annotations.*;

import java.io.Serializable;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "user")
@Inheritance(strategy = InheritanceType.JOINED)
public class UserModel implements Serializable {
    @Id
    @GeneratedValue(generator = "system-uuid")
    @GenericGenerator(name = "system-uuid", strategy = "uuid")
    private String uuid;

    @NotNull
    @Size(max = 50)
    @Column(name = "nama", nullable = false)
    private String nama;

    @NotNull
    @Size(max = 50)
    @Column(name = "username", nullable = false)
    private String username;

    @NotNull
    @Lob
    @Column(name = "password", nullable = false)
    private String password;

    @NotNull
    @Size(max = 50)
    @Column(name = "email", nullable = false)
    private String email;

    @NotNull
    @Size(max = 50)
    @Column(name = "role", nullable = false)
    private String role;
}
